### PR TITLE
nixos/none: remove

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
This windowManager and desktopManager doesn't even have
an option to use it. git history suggests to me that there's no way anyone
finds this useful anymore.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/module-list.nix [hunk 1]
```diff
@@ -906,8 +906,6 @@
   ./services/x11/window-managers/icewm.nix
   ./services/x11/window-managers/bspwm.nix
   ./services/x11/window-managers/metacity.nix
-  ./services/x11/window-managers/none.nix
-  ./services/x11/window-managers/twm.nix
```
Comment (worldofpeace) [inline nixos/modules/module-list.nix:910]: oops, will fiix

**Comment by emmanuelrosa** [comment]
This change causes an error:

```
building the system configuration...
trace: Default graphical session, 'xmonad', not found.
Valid names for 'services.xserver.displayManager.defaultSession' are:
  

trace: Default graphical session, 'xmonad', not found.
Valid names for 'services.xserver.displayManager.defaultSession' are:
  

error: The option value `services.xserver.displayManager.defaultSession' in `/etc/nixos/configuration.nix' is not of type `session name'.
(use '--show-trace' to show detailed location information)
```

I've tried using both "none+xmonad" and "xmonad" as the `defaultSession`:

```
  services.xserver = {
     enable = true;
     windowManager.xmonad = {
       enable = true;   
     };

    displayManager.defaultSession = "xmonad";
    displayManager.lightdm = {
	enable = true;
     };
  };
```

**Comment by emmanuelrosa** [comment]
I tested reverting the change and setting `defaultSession` to "none+xmonad" and it worked.

**Comment by worldofpeace** [comment]
I reverted it. I'm wondering is it that `none` gets imported default so a `none` desktop manager is somehow always an option :confused: (but it isn't a select-able?)

In that case, I should really add a comment here because the files look like lost fragments that are very often in nixpkgs.

**Comment by emmanuelrosa** [comment]
My understanding is that the selectable sessions are generated by combining every enabled window manager with every enabled desktop manager; the Cartesian product of desktop and window managers. But with the caveat that the combination of "none+none" is ignored, which is why there's no "none" session available. 
https://github.com/NixOS/nixpkgs/blob/88380453337205a65c220f68f8fed3ae1001d9f1/nixos/modules/services/x11/display-managers/default.nix#L409

**Comment by orivej** [comment]
In my case this (bisected) caused a different error: after a successful switch to the new configuration, sddm does not list the available sessions (in the lower left corner) and does not launch any session when I login (one of my users uses stock plasma5, another custom .xsession). My configuration is:

    services.xserver.displayManager.sddm.enable = true;
    services.xserver.desktopManager.plasma5.enable = true;

**Comment by arcnmx** [comment]
In my case this change made it impossible to login with lightdm because `/nix/store/XXXX-desktops/share/xsessions` was empty:
```
[+68.72s] DEBUG: Seat seat0: Failed to find session configuration xsession
[+68.72s] DEBUG: Seat seat0: Can't find session 'xsession'
```

**Comment by xaverdh** [comment]
I just stumbled upon this after my system failed to evaluate due to `sessionPackages` being empty. Maybe it should be documented in `nixos/modules/services/x11/desktop-managers/none.nix` that the [crossList logic](https://github.com/NixOS/nixpkgs/blob/252ca376fa7903f60b31498f8a41437daf7d9946/nixos/modules/services/x11/display-managers/default.nix#L402) requires it to be present?

**Comment by worldofpeace** [comment]
I will look into making it appear more idiomatic. But it's been reverted already.

**Pull request merged**